app/services/sync_provider_from_find.rb

```diff
@@ -112,7 +95,48 @@ def create_or_update_course(find_course)
       end
     end
 
-    course
+    handle_course_options_with_invalid_sites(course, find_course)
+  end
+
+  def assign_course_attributes_from_find(course, find_course)
+    course.name = find_course.name
+    course.level = find_course.level
+    course.study_mode = find_course.study_mode
+    course.description = find_course.description
+    course.start_date = find_course.start_date
+    course.course_length = find_course.course_length
+    course.recruitment_cycle_year = find_course.recruitment_cycle_year
+    course.exposed_in_find = find_course.findable?
+  end
+
+  def add_accrediting_provider(course, find_accrediting_provider)
+    if find_accrediting_provider.present?
+      accrediting_provider = Provider.find_or_initialize_by(code: find_accrediting_provider[:provider_code])
+      accrediting_provider.name = find_accrediting_provider[:provider_name]
+      accrediting_provider.save!
+
+      course.accrediting_provider = accrediting_provider
+    end
+  end
+
+  def handle_course_options_with_invalid_sites(course, find_course)
+    course_options = course.course_options.joins(:site)
+    canonical_site_codes = find_course.sites.map(&:code)
+    invalid_course_options = course_options.where.not(sites: { code: canonical_site_codes })
+
+    return if invalid_course_options.blank?
+
+    chosen_course_option_ids = ApplicationChoice.where(course_option: invalid_course_options).pluck(:course_option_id)
+
+    not_part_of_an_application = invalid_course_options.where.not(id: chosen_course_option_ids)
+    not_part_of_an_application.delete_all
+
+    part_of_an_application = invalid_course_options.where(id: chosen_course_option_ids)
+    part_of_an_application.update_all(invalidated_by_find: true)
+
+    Raven.capture_message(
+      "#{part_of_an_application.count} invalid course options chosen by candidates.",
```

Comment:
This will appear in Slack every sync for as long as invalid course options exist.

1. Too noisy?
2. Is there anything else that warrants including in this message?

Reply:
I think this is ok, and noisy is a good place to start